booster/library/Booster/Pattern/Rewrite.hs

```diff
@@ -784,15 +847,15 @@ performRewrite doTracing def mLlvmLibrary mSolver mbMaxDepth cutLabels terminalL
                     mLlvmLibrary
                     mSolver
                     simplifierCache
-                    (rewriteStep cutLabels terminalLabels pat')
+                    (withPatternContext pat' $ rewriteStep cutLabels terminalLabels pat')
                     >>= \case
                         Right (RewriteFinished mlbl uniqueId single, cache) -> do
                             whenJust mlbl $ \lbl ->
                                 emitRewriteTrace $ RewriteSingleStep lbl uniqueId pat' single
                             updateCache cache
                             incrementCounter
                             doSteps False single
-                        Right (terminal@(RewriteTerminal lbl uniqueId single), _cache) -> do
+                        Right (terminal@(RewriteTerminal lbl uniqueId single), _cache) -> withPatternContext pat' $ do
                             emitRewriteTrace $ RewriteSingleStep lbl uniqueId pat' single
```

Comment:
(nit) Maybe instead of repeating `withPatternContext pat'` , we could group the `\case` under the first one, too?
```
    withPatternContext pat' $ do
        result <- rewriteStep cutLabels ...
        case result of
            Right (RewriteFinished ...) -> do ...
```
(do-notation to avoid bracketing a humongous case expression)

Reply:
that won't work because we don't want to put `withPatternContext pat' $` around a recursive call to `doSteps`. Otherwise we will get a massive chain of `[execute][term ...]...[term ...]`, one `[term ...]` for each iteration of the loop